recipes/mpg123/all/CMakeLists.txt

```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.1)
+project(cmake_wrapper)
+
+include(conanbuildinfo.cmake)
+conan_basic_setup()
+
+if(BUILD_SHARED_LIBS)
+    set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Comment:
I would expect conan to handle this 🤔

Reply:
It just leaves CMAKE_POSITION_INDEPENDENT_CODE unchanged when `fPIC`is removed.
Maybe our current strategy to remove `fPIC` in configure is incorrect and we should do the following:
```python
def configure(self):
    if self.options.shared and self.settings.os != "Windows":
        self.options.fPIC = True
```

Reply:
🤯 ... Better to cc @conan-io/barbarians for that question

Reply:
We don't force options or settings to avoid wrong package ids according users decisions. Also, I can't remember a case where fPIC is not present when building a shared library, of course, only if the author forces a specific value, but by default I don't think we need all this strategy, the current behavior works.

Reply:
Indeed forcing options is never ok.
I looked further into this and filed a bug report + fix with mpg123:
https://sourceforge.net/p/mpg123/bugs/301/

Reply:
This is fixed not btw.
Please have look at this recipe again.